spec/operators/shareReplay-spec.ts

```diff
@@ -219,6 +219,21 @@ describe('shareReplay operator', () => {
     expectSubscriptions(source.subscriptions).toBe(sourceSubs);
   });
 
+  it('should not restart a synchronous source due to unsubscriptions if refCount is true when the source has completed', () => {
+    // The test above this one doesn't actually test completely synchronous
```

Comment:
I'm 99.9% sure that the test above doesn't have anything to do with the issue that this revert is trying to fix, because the issue with that revert only happens when `refCount` is set to `true`. So, I'm pretty sure that the test above does its job correctly.